### PR TITLE
Store: Error handling for setObject 

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Changes error handling for setObject to store. Instead of using console.error in catch, we throw Error. Secondly, we have added try-catch block to all places, where setObject is used and either  **console.error(error)** or  **appEvents.emit(AppEvents.alertError, [error])** , based on its use-case and  if it is necessity to inform user about it.

### CODE REVIEW

File: public/app/core/store.ts [hunk 1]
```diff
@@ -29,21 +29,19 @@ export class Store {
     return ret;
   }
 
-  // Returns true when successfully stored
-  setObject(key: string, value: any): boolean {
+  /* Returns true when successfully stored, thorws error if not successfully stored */
```
Comment (kaydelaney) [inline public/app/core/store.ts:32]:
```suggestion
  /* Returns true when successfully stored, throws error if not successfully stored */
```

Nit: Small typo

File: public/app/features/explore/RichHistory/RichHistory.tsx [hunk 2]
```diff
@@ -114,7 +114,14 @@ class UnThemedRichHistory extends PureComponent<RichHistoryProps, RichHistorySta
   };
 
   onSelectDatasourceFilters = (value: SelectableValue[] | null) => {
-    store.setObject(RICH_HISTORY_SETTING_KEYS.datasourceFilters, value);
+    try {
+      store.setObject(RICH_HISTORY_SETTING_KEYS.datasourceFilters, value);
+    } catch (error) {
+      console.error(error);
+    }
+    /* Set data source filters to state even though they were not succesfully saved in
```
Comment (kaydelaney) [inline public/app/features/explore/RichHistory/RichHistory.tsx:122]:
```suggestion
    /* Set data source filters to state even though they were not successfully saved in
```
Nit: Small Typo

**Review by kaydelaney — APPROVED** [review]

File: public/app/core/components/LocalStorageValueProvider/LocalStorageValueProvider.tsx [hunk 3]
```diff
@@ -24,8 +24,12 @@ export class LocalStorageValueProvider<T> extends PureComponent<Props<T>, State<
 
   onSaveToStore = (value: T) => {
     const { storageKey } = this.props;
-    store.setObject(storageKey, value);
-    this.setState({ value });
+    try {
+      store.setObject(storageKey, value);
+      this.setState({ value });
```
Comment (aocenas) [inline public/app/core/components/LocalStorageValueProvider/LocalStorageValueProvider.tsx:29]: Nitpick but the setState should probably be outside of the try catch. Not sure what can setState throw but if it throws it was not caught before.

**Review by aocenas — APPROVED** [review]
Nice work, just one small nitpick.

File: public/app/core/components/LocalStorageValueProvider/LocalStorageValueProvider.tsx (hunk 3 above)
Comment (ivanahuckova) [reply public/app/core/components/LocalStorageValueProvider/LocalStorageValueProvider.tsx]: Fixed in https://github.com/grafana/grafana/pull/23650/commits/ae118f02eb4c579f9d046069e1156d1d0903b7b0

**Pull request merged**